### PR TITLE
Collection of fixes and cleanups; including whitespace significant changes

### PR DESCRIPTION
**Pull request by travier**
This one requires careful review as it contains subtle changes due to significant whitespace changes.
Fixes: https://github.com/coreos/console-login-helper-messages/issues/102

### CODE REVIEW

**Comment by travier** [comment]
We should add an action with something like https://github.com/marketplace/actions/find-trailing-whitespace to avoid that happening in the future.

**Comment by travier** [comment]
Or rewrite that in !Bash.

File: usr/libexec/console-login-helper-messages/gensnippet_if [hunk 1]
```diff
@@ -16,38 +16,38 @@ set -e
...
-    if [ -e ${iface}/device ]; then
+    if [[ -e "${iface}/device" ]]; then
         return 0
     fi
 
-    # Check if device has any subordinate devices that are backed by a real 
+    # Check if device has any subordinate devices that are backed by a real
     # device, recursively.
-    for subifacedir in ${iface}/lower_*; do
+    for subifacedir in "${iface}/lower_"*; do
```
Comment (dustymabe) [inline usr/libexec/console-login-helper-messages/gensnippet_if:43]: does the shell glob expansion still work here?

**Review by dustymabe — APPROVED** [review]
LGTM

File: usr/libexec/console-login-helper-messages/gensnippet_if (hunk 1 above)
Comment (travier) [reply usr/libexec/console-login-helper-messages/gensnippet_if:43]: It should as I have explicitly excluded the `*` from the quotes `"`. I have not tested that however.

**Comment by travier** [comment]
CI failed two times on `ext.config.k8s.node-e2e.master`.

File: usr/libexec/console-login-helper-messages/gensnippet_if (hunk 1 above)
Comment (dustymabe) [reply usr/libexec/console-login-helper-messages/gensnippet_if:43]: might be good to run a little local test just to make sure

**Comment by dustymabe** [comment]
> CI failed two times on `ext.config.k8s.node-e2e.master`.

https://github.com/coreos/console-login-helper-messages/pull/104

**Comment by dustymabe** [comment]
A rebase should get this one passing I assume.

**Review by dustymabe — APPROVED** [review]
Still LGTM

**Comment by travier** [comment]
I'm working on testing that before we merge.

**Comment by travier** [comment]
Still need testing. I have not yet gotten to it 😕.

**Comment by travier** [comment]
Took another look at this one. The only significant commits are the last two:
- [`1d5db48` (#103)](https://github.com/coreos/console-login-helper-messages/pull/103/commits/1d5db48ca612b57b9a7297aa99360fc6e1ccf3e8)
- [`13acd55` (#103)](https://github.com/coreos/console-login-helper-messages/pull/103/commits/13acd551af1bd8c347b87061f83810920900e6b2)

The first one fixed a test that was always skipped. The fixed test will also always be skipped given that we set https://github.com/coreos/console-login-helper-messages/blob/main/usr/lib/console-login-helper-messages/libutil.sh#L12.

The second one fixes the udev action script that we currently don't use: https://github.com/coreos/console-login-helper-messages/blob/main/usr/libexec/console-login-helper-messages/gensnippet_if_udev#L18

So overall this change is a NOP for Fedora CoreOS.

Maybe we should simplify all this and remove udev support.

**Comment by dustymabe** [comment]
> Maybe we should simplify all this and remove udev support.

Yeah. The idea was that clhm would be useful for many different OS, even for ones that don't use NM, which is why we kept the udev support.

**Pull request merged**